src/huggingface_hub/hf_api.py

```diff
@@ -2245,6 +2264,674 @@ def get_full_repo_name(
+            token (`str`, *optional*):
+                An authentication token (See https://huggingface.co/settings/token)
+
+        Returns: [`DiscussionWithDetails`]
+
+        <Tip>
+
+        Raises the following errors:
+
+            - [`HTTPError`](https://2.python-requests.org/en/master/api/#requests.HTTPError)
+              if the HuggingFace API returned an error
+            - [`ValueError`](https://docs.python.org/3/library/exceptions.html#ValueError)
+              if some parameter value is invalid
+            - [`~huggingface_hub.utils.RepositoryNotFoundError`]
+              If the repository to download from cannot be found. This may be because it doesn't exist,
+              or because it is set to `private` and you do not have access.
+
+        </Tip>
+        """
+        if not isinstance(discussion_num, int) or discussion_num <= 0:
+            raise ValueError("Invalid discussion_num, must be a positive integer")
+        if repo_type not in REPO_TYPES:
+            raise ValueError(f"Invalid repo type, must be one of {REPO_TYPES}")
+        if repo_type is None:
+            repo_type = REPO_TYPE_MODEL
+        repo_id = f"{repo_type}s/{repo_id}"
+        if token is None:
+            token = HfFolder.get_token()
+
+        path = f"{self.endpoint}/api/{repo_id}/discussions/{discussion_num}"
+
+        resp = requests.get(
+            path,
+            params={"diff": "1"},
+            headers={"Authorization": f"Bearer {token}"} if token else None,
+        )
+        _raise_for_status(resp)
+
+        discussion_details = resp.json()
+        is_pull_request = discussion_details["isPullRequest"]
+
+        return DiscussionWithDetails(
+            title=discussion_details["title"],
+            num=discussion_details["num"],
+            author=discussion_details.get("author", {}).get("name", "deleted"),
+            created_at=parse_datetime(discussion_details["createdAt"]),
+            status=discussion_details["status"],
+            repo_id=discussion_details["repo"]["name"],
+            repo_type=discussion_details["repo"]["type"],
+            is_pull_request=discussion_details["isPullRequest"],
+            events=[deserialize_event(evt) for evt in discussion_details["events"]],
+            conflicting_files=discussion_details["filesWithConflicts"]
+            if is_pull_request
+            else None,
+            target_branch=discussion_details["changes"]["base"]
+            if is_pull_request
+            else None,
+            merge_commit_oid=discussion_details["changes"].get("mergeCommitId", None)
+            if is_pull_request
+            else None,
+            diff=discussion_details.get("diff"),
+        )
+
+    def create_discussion(
+        self,
+        repo_id: str,
+        title: str,
+        *,
+        token: str,
+        description: Optional[str] = None,
+        repo_type: Optional[str] = None,
+        pull_request: bool = False,
+    ) -> DiscussionWithDetails:
```

Comment:
should we have a `create_pull_request` alias that sets `pull_request=True` for DX/discoverability?

Also the docstring should mention that if you want to simply propose a file change, the easiest way is through the `create_commit(..., create_pr=True)` API